test/test_lagrangian_particle_tracking.jl

```diff
@@ -255,6 +263,44 @@ function run_simple_particle_tracking_tests(arch, timestepper; vertically_stretc
     @test all(w .≈ 0)
     @test all(s .≈ √2)
 
+    # Test active tracers
+    ts = arch_array(arch, ones(P))
+    particles = StructArray{TestParticle}((xs, ys, zs, us, vs, ws, ss, ts))
+
+    velocities = VelocityFields(grid)
+    u, v, w = velocities
+    speed = Field(√(u*u + v*v + w*w))
+    tracked_fields = merge(velocities, (; s=speed))
+    active_properties = ((property=:t, tracer=:t), )
+
+    lagrangian_particles = LagrangianParticles(particles; tracked_fields, parameters=(active_properties=active_properties, ))
+    @test lagrangian_particles isa LagrangianParticles
+
+    model = NonhydrostaticModel(grid=grid, timestepper=timestepper,
+                                velocities=velocities, particles=lagrangian_particles, tracers=(:t, ))
+
+    set!(model, u=1, v=1)
+
+    sim = Simulation(model, Δt=1e-2, stop_iteration=1)
+
+    jld2_filepath = "test_particles.jld2"
+    sim.output_writers[:particles_jld2] =
+        JLD2OutputWriter(model, (; particles=model.particles),
+                        filename=jld2_filepath, schedule=IterationInterval(1))
+
+    nc_filepath = "test_particles.nc"
+    sim.output_writers[:particles_nc] =
+        NetCDFOutputWriter(model, model.particles, filename=nc_filepath, schedule=IterationInterval(1))
+
+    sim.output_writers[:checkpointer] = Checkpointer(model, schedule=IterationInterval(1),
+                                                    dir = ".", prefix = "particles_checkpoint")
+
+    model = sim.model
+    run!(sim)
+
+    ni, nj, nk, d = Oceananigans.LagrangianParticleTracking.getmask(0.0, 0.0, 0.5, grid, (Center(), Center(), Center()))
+    vols = [Oceananigans.Operators.volume(i, nj[idx], nk[idx], grid, Center(), Center(), Center()) for (idx, i) in enumerate(ni)]
+    @test sum(model.tracers.t) ≈ (sum(1e-2*P ./vols) / 8)
```

Comment:
What's being tested here? I don't understand this, and the `1e-2` and `8` need to be explained.

Reply:
This test checks that the correct total change is made to the tracer concentration, and each particle is adding 1 unit/s for `1e-2`s spreading spread over the volume of the 8 nearest cells.

I'll add another test I had in my other implementation that checks that the mask function is selecting the correct cells too.

Reply:
Something just like

```julia
Ncells  = 8
tolerance = 1e-2
total_volume = [Oceananigans.Operators.volume(i, nj[idx], nk[idx], grid, Center(), Center(), Center()) for (idx, i) in enumerate(ni)]
average_P = sum(P ./ (Ncells * total_volume))
@test sum(model.tracers.t) ≈ tolerance * average_P
```

could help. But I think I got the names wrong above, and I'm not sure why we compare `t` to `P`.

Reply:
Okay I'll write it out more explicitly like that to add some clarity. Here `P` is the number of particles and `1e-2`is the amount each particle should have reduced the amount of tracer by.